### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by Leminode**
Implemented logging to file using NLog, added exception handling, other minor fixes.

### CODE REVIEW

File: Energy Saver/Pages/Delete.cshtml.cs [hunk 1]
```diff
@@ -33,24 +33,32 @@ public async Task<IActionResult> OnGetAsync(int? id)
         {
             if (id == null || _context.Taxes == null)
             {
-                return RedirectToPage("./Index");
```
Comment (priolette) [inline Energy Saver/Pages/Delete.cshtml.cs:36]: You suggested me to delete NotFound() references, you should too ;)

File: Energy Saver/Pages/Statistics.cshtml.cs [hunk 2]
```diff
@@ -12,20 +12,27 @@ namespace Energy_Saver.Pages
 {
     public class StatisticsModel : PageModel
     {
+        public delegate void StatisticsTaxesHandler(object source, NotificationService.NotificationArgs args);
```
Comment (priolette) [inline Energy Saver/Pages/Statistics.cshtml.cs:15]: Why not EventHandler class?

File: Energy Saver/Pages/Delete.cshtml.cs (hunk 1 above)
Comment (Leminode) [reply Energy Saver/Pages/Delete.cshtml.cs:36]: I don't think we should get rid of the not found all together, it's good in some scenarios.

File: Energy Saver/Pages/Statistics.cshtml.cs (hunk 2 above)
Comment (Leminode) [reply Energy Saver/Pages/Statistics.cshtml.cs:15]: Will change later

**Review by Tauras-Rim — APPROVED** [review]
Looks great, good job

**Pull request merged**